clc/modules/object-storage-common/src/main/java/com/eucalyptus/objectstorage/msgs/ObjectStorage.groovy

```diff
@@ -242,6 +279,25 @@ public class ObjectStorageErrorMessageType extends BaseMessage {
   }
 }
 
+public class ObjectStorageErrorMessageExtendedType extends ObjectStorageErrorMessageType {
+  String method;
+
+  def ObjectStorageErrorMessageExtendedType() {}
+
+  def ObjectStorageErrorMessageExtendedType(String message,
+  String code,
+  HttpResponseStatus status,
+  String resourceType,
+  String resource,
+  String requestId,
+  String hostId,
+  BucketLogData logData,
+  String method) {
+    super(message,code,status,resourceType,resource,requestId,hostId,logData);
+    this.method = method;
+  }
+}
+
```

Comment:
Is there a necessity for an ObjectStorageErrorMessageExtendedType? Why not add the method to ObjectStorageErrorMessageType class and create a new constructor, one that takes in method as an argument in addition to others? This might save cycles spent on instanceOf and if-else processing at a lot of places.